lib/foreman_inventory_upload/generators/json_stream.rb

```diff
@@ -26,8 +26,14 @@ def raw(string)
         @out << string
       end
 
-      def simple_field(name, value, last = false)
-        @out << "\"#{name}\": #{stringify_value(value)}#{last ? '' : ','}" unless value.nil?
+      def simple_field(name, value, last = false, &block)
+        return if value.nil?
+        return if value.kind_of?(String) && value.empty?
+        return if value.kind_of?(Array) && value.compact.empty?
```

Comment:
```suggestion
        return if value.empty?
        return if value.kind_of?(Array) && value.compact.empty?
```

`empty?` works on simple values.

Reply:
yes but for numbers it can return an error as there is no `empty?` for `Integer`
in case we will want to pass in an Integer (?)

Reply:
You are right, sorry:
``` ruby
return if value.try(:empty?)
```
will do the trick for types that don't have empty, like Integer.

Reply:
Thanks, updated !